nomos-services/cryptarchia-consensus/src/lib.rs

```diff
@@ -553,29 +695,51 @@ where
         blob_selector: BS,
         cl_mempool_relay: MempoolRelay<ClPool::Item, ClPool::Item, ClPool::Key>,
         da_mempool_relay: MempoolRelay<DaPoolAdapter::Payload, DaPool::Item, DaPool::Key>,
+        sampling_relay: SamplingRelay<SamplingBackend::BlobId>,
     ) -> Option<Block<ClPool::Item, DaPool::Item>> {
         let mut output = None;
         let cl_txs = get_mempool_contents(cl_mempool_relay);
         let da_certs = get_mempool_contents(da_mempool_relay);
-
-        match futures::join!(cl_txs, da_certs) {
-            (Ok(cl_txs), Ok(da_certs)) => {
+        let blobs_ids = get_sampled_blobs(sampling_relay);
```

Comment:
So this is the proposer who is querying for sampled blobs, not the mempool, right? Does it actually make any difference here?

Reply:
The mempool will trigger the sampling, but it has no usage for the already sampled blobs.
Btw, mempools will change once we actually have TXs, as blobs are part of the TXs themselves. Probably DA mempool gonna be remove entirely.